service/src/test/java/uk/gov/hmcts/reform/fpl/service/ApplicationDocumentsServiceTest.java

```diff
@@ -0,0 +1,151 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.fpl.model.ApplicationDocument;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.Document;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+import uk.gov.hmcts.reform.fpl.utils.DocumentUploadHelper;
+import uk.gov.hmcts.reform.fpl.utils.FixedTimeConfiguration;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.caseData;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.emptyCaseData;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {
+    JacksonAutoConfiguration.class,
+    ApplicationDocumentsService.class,
+    FixedTimeConfiguration.class
+})
+class ApplicationDocumentsServiceTest {
+
+    private static final String HMCTS_USER = "HMCTS";
+    private static final String LA_USER = "kurt@swansea.gov.uk";
+    private static final String OLD_FILENAME = "Old file";
+    private static final String NEW_FILENAME = "New file";
+    private static final LocalDateTime PAST_DATE = LocalDateTime.now().minusDays(2);
+    private static final UUID DOCUMENT_ID = UUID.randomUUID();
+
+    @Autowired
+    private Time time;
+
+    @Autowired
+    private ApplicationDocumentsService applicationDocumentsService;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @MockBean
+    private DocumentUploadHelper documentUploadHelper;
+
+    @BeforeEach
+    void setup() {
+        when(documentUploadHelper.getUploadedDocumentUserDetails()).thenReturn(HMCTS_USER);
+    }
+
+    @Test
+    void shouldSetUploadedByAndDateTimeOnNewCaseDocument() {
+        CaseData caseData = caseData();
+
+        List<Element<ApplicationDocument>> previousDocuments = emptyCaseData().getDocuments();
+        List<Element<ApplicationDocument>> documents = caseData.getDocuments();
+
+        Map<String, Object> map = applicationDocumentsService.updateCaseDocuments(documents, previousDocuments);
+
+        assertThat(map.get("documents")).isEqualToComparingOnlyGivenFields(Document.builder()
+            .uploadedBy(HMCTS_USER)
+            .dateTimeUploaded(time.now()));
```

Comment:
Looks to be used in a couple of assertions here, could extract to a method like `Document buildDocument(String uploadedBy, LocalDateTime timeOfUpload)`